Record fetch requests in breadcrumb

romainneutron opened the pull request:
Hello,

this is an integration of capturing fetch requests the same way XMLHttpRequests are captured for breadcrumb.

Tests are passing on my browser using `node_modules/.bin/grunt run:test` on `http://localhost:8000/test/integration/`
However, running `node_modules/.bin/grunt test` fails locally with

```
Running "mocha:integration" (mocha) task
Testing: test/integration/index.html

․․․․․․․․․․․․․․․․XMLHttpRequest cannot load https://example.com/api/1/store/?sentry_key=public. Origin file:// is not allowed by Access-Control-Allow-Origin.
․․․․․․․․․․․

  26 passing (5s)
  1 failing

  1) integration breadcrumbs should record a fetch request:
     timeout of 2000ms exceeded
```

I don't know why. If anyone has an idea...

benvinegar commented:
Sweet! I'll probably merge this as-is, but follow-up and change the config options so that `xhr` becomes `http` and includes both `fetch` and `XMLHttpRequest`.

benvinegar commented:
> I don't know why. If anyone has an idea...

The tests are run in Phantom (1?), which probably doesn't have `fetch`. It likely errors (method does not exist) and never completes, hence the timeout.

romainneutron commented:
@benvinegar I just updated the PR and fixed the tests. I added the fetch polyfill to the test iframe but forgot the es6-promise polyfill to get it work correctly :/ So tests are now passing

About renaming fetch+xhr to http setting, are you sure this is a good idea? Sounds like a BC break for users. Anyway, if you're sure, I can make the change in this PR. Let me know

benvinegar commented:
> About renaming fetch+xhr to http setting, are you sure this is a good idea? Sounds like a BC break for users.

We would make it backwards compatible.

For now, let's just keep the option "xhr" ... and have it encompass both `XMLHttpRequest` and `fetch`. That way, if someone has explicitly disabled the "xhr" option, they won't magically have "fetch" requests appear in their breadcrumbs when they upgrade to the next version.

romainneutron commented:
PR updated, should be alright now :)

romainneutron commented:
Anything left here?
